nimterop/getters.nim

```diff
@@ -306,4 +316,26 @@ proc getNimExpression*(expr: string): string =
 
 proc getSplitComma*(joined: seq[string]): seq[string] =
   for i in joined:
-    result = result.concat(i.split(","))
\ No newline at end of file
+    result = result.concat(i.split(","))
+
+proc dll*(path: string): string =
+  let
+    (dir, name, _) = path.splitFile()
+
+  result = dir / (DynlibFormat % name)
+
+proc loadPlugin*(fullpath: string) =
+  doAssert fileExists(fullpath), "Plugin file does not exist: " & fullpath
+
+  let
+    pdll = fullpath.dll
+  if not fileExists(pdll) or
+    fullpath.getLastModificationTime() > pdll.getLastModificationTime():
+    discard execAction("nim c --app:lib " & fullpath)
+  doAssert fileExists(pdll), "No plugin binary generated for " & fullpath
+
+  let lib = loadLib(pdll)
+  doAssert lib != nil, "Plugin $1 compiled to $2 failed to load" % [fullpath, pdll]
+
+  gStateRT.onSymbol = cast[typeof(gStateRT.onSymbol)](lib.symAddr("onSymbol"))
```

Comment:
like u said in gitter that shd that be `type()` ?

Reply:
I fixed it in latest push

Reply:
[RESOLVED] (btw feel free to directly just mark as resolved when u resolve :) )